### PR TITLE
make delayed progress tests more resilient

### LINKED ISSUE #3520: TestProgress test is flaky in CI, sometimes produces more '.' than expected
## Flaky Test

* **Test Name:** TestProgress
* **Link:** https://github.com/elastic/elastic-agent/blob/8bf732e097e42e736988b49155483ef4110889d4/internal/pkg/agent/install/progress_test.go#L79
* **Branch:** main
* **Artifact Link:** 
* **Notes:** In CI the test can take longer so you get more '.' than expected.  Just changing the regext from `.\{3}` to `.\{3,}` should fix the issue.  That way we check for at least `...` but more is fine if CI takes longer to run the test.
* 

### Stack Trace

```
=== FAIL: internal/pkg/agent/install TestProgress/nested_step_delayed_success (0.09s)
    progress_test.go:136: 
        	Error Trace:	/Users/runner/work/elastic-agent/elastic-agent/internal/pkg/agent/install/progress_test.go:136
        	Error:      	Expect "step starting...
        	            	   substep 1 starting... DONE
        	            	   substep 2 starting...... DONE
        	            	   DONE
        	            	" to match "step starting\.{3}\n   substep 1 starting\.{3}\.+ DONE\n   substep 2 starting\.{3}\.+ DONE\n   DONE\n"
        	Test:       	TestProgress/nested_step_delayed_success


```

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?

allow progress tests to take more time than expected for CI

## Why is it important?

tests are flaky in CI without it

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Author's Checklist

- [ ]

## Related issues

- Closes #3520 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: internal/pkg/agent/install/progress_test.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func TestProgress(t *testing.T) {
 
 		rs.Failed()
 
-		require.Equal(t, "step 1 starting... FAILED\n", string(w.buf))
+		require.Regexp(t, regexp.MustCompile(`step 1 starting.{3,} FAILED\n`), string(w.buf))
```
Comment (ycombinator) [inline internal/pkg/agent/install/progress_test.go:113]: Are we sure this one needs to be changed? There is no delay in this test between the step starting and failing so we should expect to see exactly three `.`s.

Comment (leehinman) [reply internal/pkg/agent/install/progress_test.go:113]: I'm being paranoid, the Go scheduler could stop your thread of execution between the `StepStart` and `Failed`, then OS scheduler puts your process to sleep, and the container run time halts your container for 10 seconds, when it all comes back the ticker has gone off and you get more "." than you expected.

Comment (ycombinator) [reply internal/pkg/agent/install/progress_test.go:113]: Fair enough.

**Review by ycombinator — APPROVED** [review]

**Comment by cmacknz** [comment]
Test failures are unrelated, force merging so we can stop having this flaky test block other PRs.

**Pull request merged**